pkg/api/routes.go

```diff
@@ -40,16 +41,25 @@ const (
 )
 
 type RouteHandler struct {
-	c *Controller
+	c        *Controller
+	blobLock sync.Mutex
```

Comment:
yup, this is what I have in my PR as well, except just did a defer unlock() pattern.
A manifestLock is also needed coz multiple clients can potentially update/delete a manifest, and we need to decide a lock order in that case - manifest lock followed by blob lock.

Reply:
I'm not sure what defer unlock() does in terms of helping us out minimizing the critical section?